src/main/java/com/openlattice/edm/type/PropertyType.java

```diff
@@ -89,6 +83,10 @@ public PropertyType(
         this.piiField = piiField.orElse( false );
         this.multiValued = multiValued.orElse( true );
         this.analyzer = analyzer.orElse( Analyzer.STANDARD );
+
+        if ( EdmPrimitiveTypeKind.Binary == this.datatype && postgresIndexType.isPresent() ) {
+            checkArgument( postgresIndexType.get() == IndexType.NONE, "Indexes are not allowed on Binary datatypes" );
```

Comment:
If postgresIndexType isn't present and datatype is binary, postgresIndexType will be set to BTREE

Reply:
Yeah I was just thinking this. I'll make another PR